src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -0,0 +1,236 @@
+     * <p>
+     * It first checks whether the current node is bootstrapped.
+     * If not, it continues checking in intervals of 1 second.
+     * If yes, it sets up the corfuRuntime and continues execution
+     * of the policy.
+     * <p>
+     * It executes the policy which detects and reports failures.
+     * Once detected, it triggers the trigger handler which takes care of
+     * dispatching the appropriate handler.
+     * <p>
+     * Currently executing the periodicPollPolicy.
+     * It executes the the polling at an interval of every 1 second.
+     * After every poll it checks for any failures detected.
+     */
+    private void taskScheduler() {
+
+        Layout currentLayout;
+        LayoutView layoutView;
+        CorfuRuntime corfuRuntime;
+
+        do {
+            corfuRuntime = getCorfuRuntime();
+            if (isShutdown) return;
+        } while (corfuRuntime == null);
+
+
+        while (true) {
+            try {
+                corfuRuntime.invalidateLayout();
+
+                // Fetch the latest layout view through the runtime.
+                layoutView = corfuRuntime.getLayoutView();
+                currentLayout = layoutView.getLayout();
+                //TODO: Store this layout on disk for persistence.
+
+                // Execute the failure detection policy once.
+                failureDetectorPolicy.executePolicy(currentLayout, corfuRuntime);
+
+                // Get the server status from the policy and check for failures.
+                HashMap<String, Boolean> map = failureDetectorPolicy.getServerStatus();
+                if (!map.isEmpty()) {
+                    // Failure present. Triggering the handler.
+                    triggerHandlerDispatcher(currentLayout, map.keySet());
+                }
+
+                Thread.sleep(policyExecuteInterval);
+
+            } catch (InterruptedException err) {
+                log.debug("Sleep interrupted.");
+                if (isShutdown) {
+                    return;
+                }
+            }
+        }
+    }
+
+    /**
+     * Shuts down the fault detector service.
+     */
+    public void shutdown() {
+        isShutdown = true;
+
+        // Shut down the Corfu Runtime.
+        if (corfuRuntime != null) {
+            corfuRuntime.isShutdown = true;
+            if (corfuRuntime.layout != null) {
+                corfuRuntime.layout.cancel(true);
+            }
+        }
+
+        // Shutting the fault detector.
+        schedulerService.shutdownNow();
+        while (!schedulerService.isShutdown()) {
+            schedulerService.shutdownNow();
+        }
+    }
+
+    /**
+     * Triggers the handler dispatcher.
+     *
+     * @param layout        Current layout with failed nodes.
+     * @param failedServers Set of server nodes to be removed from the layout.
+     */
+    private void triggerHandlerDispatcher(Layout layout, Set<String> failedServers) {
```

Comment:
Since this isnt an interface or abstract, then just add it when you want to implement it. There's no point of having this stub.